managedplugin/plugin.go

```diff
@@ -423,15 +423,37 @@ func getFreeTCPAddr() (string, error) {
 }
 
 func (c *Client) startLocal(ctx context.Context, path string) error {
-	if c.useTCP {
-		tcpAddr, err := getFreeTCPAddr()
-		if err != nil {
-			return fmt.Errorf("failed to get free port: %w", err)
-		}
-		c.tcpAddr = tcpAddr
-		return c.startLocalTCP(ctx, path)
-	}
-	return c.startLocalUnixSocket(ctx, path)
+	attempt := 0
+	return retry.Do(
+		func() error {
+			attempt++
+			c.logger.Debug().Str("path", path).Int("attempt", attempt).Msg("starting plugin")
```

Comment:
We can also customize which error we should retry on via `retry.RetryIf()` but I don't think we have enough information to classify the failures.
We already check the binary exists on local registry, and for remote registries (GitHub, CloudQuery) it should fail before trying to launch if something is wrong